tests/e2e/testdata/pods.yaml

```diff
@@ -0,0 +1,75 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: pendingpod2
+  name: pendingpod2
+  namespace: default
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: pendingpod2
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: pendingpod2
+    spec:
+      nodeSelector:
+        debug-node: "non"
+      containers:
+      - command:
+        - /agnhost
+        - netexec
+        - --http-port=8080q
+        image: registry.k8s.io/e2e-test-images/agnhost:2.39
```

Comment:
where is this image from? will the cluster attempt to download it?

Reply:
> where is this image from?

The question is a bit unclear but, [kubernetes/kubernetes (/test/images/agnhost)](https://github.com/kubernetes/kubernetes/tree/master/test/images/agnhost#image) seams to be the source.

> will the cluster attempt to download it?

Again, the question is a bit unclear but, no end-user clusters will not download this; it is just used for e2e testing AFAIK.

Reply:
Apologies for not being precise, the reason I was asking mainly is:
* Will this image be downloaded to a node? (= can I run e2e tests in an airgapped environment, or when developing on a plane?)
* Is the image source stable (= will CI integration tests break)?

Reply:
I believe the image is stable ([scheduled releases](https://github.com/kubernetes/kubernetes/commits/master/test/images/agnhost/VERSION)), and safe for us to use in the given context. LMK if I missed anything though.